src/python/WMCore/MicroService/Service/Data.py

```diff
@@ -113,6 +119,11 @@ def post(self):
             if 'request' in data:
                 reqName = data['request']
                 result = self.mgr.info(reqName)
+            if cherrypy.request.headers.get('Accept-Encoding', '') == 'gzip':
```

Comment:
Given that this might be used in different parts of the code, I would suggest to have a specific function for this check. HTTP headers are tricky as well, and I have seen implementations of it either in case sensitive or not.

I don't want to over-complicate this logic, but we need to make sure which case we are dealing with and maybe add a comment as well

Reply:
I doubt that additional wrap of already existing object is required. It will be over-complicated logic. I will leave it as is.

Reply:
Well, you just duplicated this code. I would suggest something like:
```
if acceptGzip(blah):
    return compressResponse(blah)
```

this way it will be easier to make unit tests for these specific methods as well, in addition to keeping a clear separation of the code.